File: src/sbt-test/sbt-reactive-app/detection/lagom-scala-api/src/main/scala/left/SimpleService.scala
package left

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.transport.Method
import com.lightbend.lagom.scaladsl.api.{ Descriptor, Service, ServiceCall }

trait SimpleService extends Service {
  def simple(input: String): ServiceCall[NotUsed, String]

  override final def descriptor: Descriptor = {
    import Service._
    named("left")
      .withCalls(restCall(Method.GET, "/left/:text", simple _))
      .withAutoAcl(true)
  }
}

File: src/sbt-test/sbt-reactive-app/detection/lagom-java-api/src/main/java/right/SimpleService.java
package right;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import static com.lightbend.lagom.javadsl.api.Service.*;

public interface SimpleService extends Service {

  ServiceCall<NotUsed, String> simple(String input);

  @Override
  default Descriptor descriptor() {
    // @formatter:off
    return named("right").withCalls(
        pathCall("/right/:text", this::simple)
      ).withAutoAcl(true);
    // @formatter:on
  }
}
